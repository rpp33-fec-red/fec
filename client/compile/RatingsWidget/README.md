# RED ATELIER

## Table of Contents
- [Overview](#overview)
- [Description](#description)
- [Getting Started](#getting-started)
- [File Structure](#file-structure)

## Overview
Red Atelier is a front end redesign of an retail application, built with React.

## Description
Red Atelier

- [Product Overview](#product-overview)
- [Related Products & My Outfit](#related-products-&-your-outfit)
- [Questions & Answers](#questions-&-answers)
- [Ratings & Reviews](#ratings-&-reviews)

### Product Overview
_Developed by Grant Mitchell_

### Related Products & Your Outfit
_Developed by Thao Nguyen_

The Related Product presents the users with a set of products that are similar to the currently selected product, provided by the external API.

This section shows a list of product cards that sit on a horizontally scrolling carousel. The amount of cards showed depends on the current screen width. If there is more cards to show, right arrow will appear to let user click on it. When reaching the end of the list, left arrow will appear to let user going back to the previous set.

![products-carousel](https://user-images.githubusercontent.com/88979402/157744146-2a1eace6-4ba9-4b8c-95e1-7e8edc680a18.gif)


Each product card displays the essential information for a single product such as category, name, price, photo and ratings. The card is clickable. Clicking the card will navigate to the detail page for that product. Consequently, a new set of products relating to the clicked product will now be displayed.

![products click](https://user-images.githubusercontent.com/88979402/157744177-c32a0ab7-19ea-4a77-9f2b-7beeecda76ee.gif)


Each product card has a star icon in the upper righthand corner, that allows user to compare the current product with the clicked related product. A modal will pop up and display a table with the features line up for both products.

![compare modal](https://user-images.githubusercontent.com/88979402/157744208-22dbc2b8-c5d8-4b2e-826f-6046f0c87e97.gif)

Your Outfit provides users a way to select and group products together as an outfit. This list has the same format as the related products section. If users want to add the current product to their outfit, they can select the "+" button. To remove an outfit item, users can simply click the X on the outfit card. Product can only be add to Outfit list once.

Each customer will have their own outfit list. The list will remain the same regardless of which product detail page they are viewing. The list is also persistent, allowing user to exit and return to the same list at later time.

![outfit list](https://user-images.githubusercontent.com/88979402/157753274-7b09df39-971a-488c-959d-0f166b00d66d.gif)


### Questions & Answers
_Developed by Nick Gerrard_

### Ratings & Reviews
_Developed by Alysha Gilliard_
The Ratings and Reviews widget allows users to view and add new reviews to the selected product.

This widgets has the following features:

Write new review
Reviews List
Sorting
Rating Breakdown
Product Breakdown



## Getting Started
This project can be run by executing the following steps:
### Installation
1. Download or clone this git repository onto your local machine
2. Navigate to the root directory and run `npm install`

### Build
3. In same directory, run `npm run build`

### Start Server
4. In same directory, run `npm run start`

## File Structure
- Client [Folder]
    - Compile [Folder]
        - Overview [Foler]
        - Questions [Folder]
        - RatingsWidget [Folder]
        - RelatedItems [Folder]
        - app.js
    - Public [Folder]
        - index.html
- Server [Folder]
    - server.js
- .gitignore
- (Config Files)
- package.json
- package-lock.json
