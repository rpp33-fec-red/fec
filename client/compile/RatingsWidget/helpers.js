const validateFields = (formData, applicableCharacteristics, rating, files) => {
  let success = true;
  let formFields = {};
  const mandatoryFields = ['body', 'summary', 'email', 'nickname'];

  for (const [key, value] of formData.entries()) {
    formFields[key] = value;
  }

  let errorMessage = 'You must enter the following:';

  const addToErrorMessage = (message) => {
    if (success === true) {
      success = false;
    }
    const lastCharacter = errorMessage.charAt(errorMessage.length - 1);
    if (lastCharacter === ':') {
      errorMessage += ' ' + message;
      return;
    }
    errorMessage += ', ' + message;
  };

  // checks if all mandatory fields (expect for characteristics) are filled out and compiles error message
  for (const field in formFields) {

    if (mandatoryFields.indexOf(field) !== -1 && formFields[field].length === 0) {
      addToErrorMessage(field);
    }
  }

  // checks if all characteristic fields are filled out and compiles error message
  for (var characteristic in applicableCharacteristics) {
    const characteristicId = applicableCharacteristics[characteristic].id;
    if (!formFields[characteristicId]) {
      addToErrorMessage(characteristic.toLowerCase());
    }
  }

  // checks if review body is less than 50 characters
  if (formFields.body.length < 50 && errorMessage.indexOf('body') === -1) {
    const characterCountError = 'more than 50 characters in your review body';
    addToErrorMessage(characterCountError);
  }

  // checks if email address is valid
  if (!formFields.email.includes('@') && !formFields.email.includes('.com') && errorMessage.indexOf('email') === -1) {
    const characterCountError = 'valid email address';
    addToErrorMessage(characterCountError);
  }

  // if no star rating provided
  if (rating === 0) {
    addToErrorMessage('rating');
  }

  // image count validatation [in-progress]
  if (files.length > 5) {
    addToErrorMessage('5 or less photos');
  }

  if (success) {
    const successMessage = 'Your review has been submitted.';
    return successMessage;
  } else {
    return errorMessage;
  }

};

const formatReviewData = (fields, applicableCharacteristics, starRating, product_id) => {

  // Converts characteristic rating details into format that works with the API ("rating_id": rating - ex: {"14": 5, "15": 5 //...})
  let reviewCharRating = {};
  for (var characteristic in applicableCharacteristics) {
    const characteristicRating_id = applicableCharacteristics[characteristic].id;
    const characteristicRating = event.target.elements[characteristic].value;
    reviewCharRating[characteristicRating_id] = parseInt(characteristicRating);
  }

  const recommend = fields.recommend.value === 'true' ? true : false;
  const summary = fields.summary.value;
  const body = fields.body.value;
  const name = fields.nickname.value;
  const email = fields.email.value;

  const reviewData = {
    product_id: product_id,
    rating: starRating,
    summary: summary,
    body: body,
    recommend: recommend,
    name: name,
    email: email,
    photos: [''],
    characteristics: reviewCharRating
  };

  return reviewData;
};

const getCharacteristicsDescriptions = (characteristic) => {
  let valueDescriptions;
  switch(characteristic){
  case 'Size':
    valueDescriptions = {
      1: 'A size too small',
      2: '½ a size too small',
      3: 'Perfect',
      4: '½ a size too big',
      5: 'A size too wide'
    };
    break;
  case 'Width':
    valueDescriptions = {
      1: 'Too narrow',
      2: 'Slightly narrow',
      3: 'Perfect',
      4: 'Slightly wide',
      5: 'Too wide'
    };
    break;
  case 'Comfort':
    valueDescriptions = {
      1: 'Uncomfortable',
      2: 'Slightly uncomfortable',
      3: 'Ok',
      4: 'Comfortable',
      5: 'Perfect'
    };
    break;
  case 'Quality':
    valueDescriptions = {
      1: 'Poor',
      2: 'Below average',
      3: 'What I expected',
      4: 'Pretty great',
      5: 'Perfect'
    };
    break;
  case 'Length':
    valueDescriptions = {
      1: 'Runs Short',
      2: 'Runs slightly short',
      3: 'Perfect',
      4: 'Runs slightly long',
      5: 'Runs long'
    };
    break;
  case 'Fit':
    valueDescriptions = {
      1: 'Runs tight',
      2: 'Runs slightly tight',
      3: 'Perfect',
      4: 'Runs slightly long',
      5: 'Runs long'
    };
    break;
  }
  return valueDescriptions;
};

const calculateRatingMetrics = (ratings) => {
  let weightedSum = 0;
  let count = 0;
  let ratingsPercentage = {};

  for (let val in ratings) {
    weightedSum += val * ratings[val];
    count += parseInt(ratings[val]);
  }

  for (let val in ratings) {
    ratingsPercentage[val] = (parseInt(ratings[val]) / count) * 100;
  }

  const averageRating = (Math.round((weightedSum/count) * 4) / 4).toFixed(2);

  const ratingsMetrics = {
    averageRating: averageRating,
    ratingsPercentage: ratingsPercentage
  };

  return ratingsMetrics;
};

export {validateFields, formatReviewData, getCharacteristicsDescriptions, calculateRatingMetrics};
