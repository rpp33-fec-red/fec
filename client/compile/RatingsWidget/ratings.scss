.ratings-and-reviews {
  display: flex;
  margin: 30px;

  .reviews {
    flex: 1;

    .reviews-list {
      max-height: 80vh;
      overflow: auto;
      .review-tile {

        margin: 30px;
        border-bottom: 1px solid black;

        .review-header {
          display: flex;
          flex-direction: row;

          .star-rating {
            width: 50%;
          }
          .right-corner {
            text-align: right;
            flex: 1;
          }
        }

        .review-main {
          .thumbnail-display {
            display: flex;
            flex-direction: row;
            gap: 10px;

            .review-image {
              max-width: 100px;
              height: auto;
            }
          }
        }
        .review-response {
          background-color:rgb(206, 206, 206);
          padding: 10px;

          p,h4 {
            margin: 5px;
          }
        }
      }
    }
  }

  .ratings {
    width: 25%;
  }
}

.modal-window {
  position: fixed;
  z-index: 1;
  left: 0;
  top: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.4);
  padding-top: 5vh;
  overflow: auto;
}

.modal-content {
  background-color: white;
  display: block;
  margin: auto;
  max-width: 60%
}

.form-content {
  display: flex;
  .div {
    justify-content: center;
  }
  flex-direction: column;
  padding: 50px;
}

.star-rating-input {
  display: flex;
  flex-direction: row;
}

.radio-buttons {
  display: grid;
  grid-template-columns: repeat(8, 1fr);
  gap: 10px;
  align-items: end;
}

.ratings-breakdown-bar {
  display: flex;
  flex-direction: row;
}


