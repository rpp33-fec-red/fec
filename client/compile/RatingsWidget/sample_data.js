var reviewsData = {
  "product": "2",
  "page": 0,
  "count": 5,
  "results": [
    {
      "review_id": 5,
      "rating": 3,
      "summary": "I'm enjoying wearing these shades",
      "recommend": true,
      "response": null,
      "body": "Comfortable and practical.",
      "date": "2019-04-14T00:00:00.000Z",
      "reviewer_name": "shortandsweeet",
      "helpfulness": 5,
      "photos": [{
          "id": 1,
          "url": "https://images.unsplash.com/photo-1560570803-7474c0f9af99?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=975&q=80"
        },
        {
          "id": 2,
          "url": "https://images.unsplash.com/photo-1560570803-7474c0f9af99?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=975&q=80"
        },
        // ...
      ]
    },
    {
      "review_id": 3,
      "rating": 4,
      "summary": "I am liking these glasses",
      "recommend": false,
      "response": "Glad you're enjoying the product!",
      "body": "They are very dark. But that's good because I'm in very sunny spots",
      "date": "2019-06-23T00:00:00.000Z",
      "reviewer_name": "bigbrotherbenjamin",
      "helpfulness": 5,
      "photos": [],
    },
    {
      "review_id": 4,
      "rating": 3,
      "summary": "I'm enjoying wearing these shades because they block the sun from my eyes and they are stylish.",
      "recommend": true,
      "response": null,
      "body": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras eu elementum ligula, in feugiat purus. Aenean sed nisl imperdiet, ultrices neque sit amet, vehicula lorem. Duis ut gravida dolor, ut ultricies nunc. Pellentesque ultrices efficitur elementum. Vestibulum et lacinia diam, ac hendrerit lacus. Nam a justo vitae libero dictum rhoncus id eu lectus. Donec aliquam elit nec massa vulputate, sit amet bibendum nunc varius. Aliquam sed tristique lectus, vitae rutrum lacus. Quisque ut nibh interdum felis porttitor ultricies eget auctor nunc. Vivamus vestibulum laoreet justo. Morbi iaculis nibh eu risus fermentum, id scelerisque ex hendrerit. Aliquam dui ex, blandit vitae suscipit vitae, lobortis vel enim. Donec lorem nisi, luctus vitae enim non, volutpat bibendum leo. Sed id augue eget tortor accumsan imperdiet. Duis at faucibus nulla. Proin volutpat massa ut velit suscipit, et faucibus tortor consequat. Vivamus maximus quam quis neque mattis iaculis. Vestibulum sed condimentum nibh, rutrum rhoncus nisl. Nulla condimentum pellentesque sem sed maximus. Donec cursus purus arcu, ac malesuada velit facilisis pulvinar. Donec non massa vestibulum, suscipit enim quis, vestibulum nisl. In consequat ex sed vulputate accumsan. Integer nec velit vitae nulla tempor aliquam. In vulputate mi quis ligula pharetra, eu pharetra tellus venenatis. Quisque gravida nibh a lorem commodo ornare a vitae lacus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Ut molestie ante urna, nec bibendum augue varius sit amet. Sed vitae consectetur tellus, vel congue ex. Sed bibendum pellentesque scelerisque. Nulla vitae sodales sem. Nulla volutpat nulla et enim dapibus sollicitudin. Proin consectetur sodales dapibus. Nullam sagittis sem nisi, at condimentum metus interdum id.",
      "date": "2020-07-24T05:00:00.000Z",
      "reviewer_name": "alyshagilliard",
      "helpfulness": 8,
      "photos": [{
          "id": 1,
          "url": "https://images.unsplash.com/photo-1560570803-7474c0f9af99?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=975&q=80"
        },
        {
          "id": 2,
          "url": "https://images.unsplash.com/photo-1560570803-7474c0f9af99?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=975&q=80"
        },
        {
          "id": 3,
          "url": "https://images.unsplash.com/photo-1560570803-7474c0f9af99?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=975&q=80"
        }

      ]
    }
  ]
}

export default reviewsData;
