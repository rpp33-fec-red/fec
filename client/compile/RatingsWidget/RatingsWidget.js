import React from 'react';
import './ratings.scss';
import Reviews from './Reviews/Reviews.js';
import Ratings from './Ratings/Ratings.js';
import PropTypes from 'prop-types';

class RatingsWidget extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      sortedBy: 'relevant',
      filteredBy: {},
      reviews: [],
      reviewsMetadata: {},
      recommendPercentage: 0,
      ratingsMetrics: {}
    };
    this.getReviews = this.getReviews.bind(this);
    this.updateSorting = this.updateSorting.bind(this);
    this.getReviewsMetadata = this.getReviewsMetadata.bind(this);
    this.getRecommendedPercentage = this.getRecommendedPercentage.bind(this);
    this.updateRatingFilter = this.updateRatingFilter.bind(this);
    this.removeRatingFilter = this.removeRatingFilter.bind(this);
  }

  componentDidMount () {
    this.getReviews();
    this.getReviewsMetadata();
  }


  getReviews() {
    const that = this;
    const count = 50;
    this.props.getReviews([`reviews?product_id=${this.props.product_id}%26sort=${this.state.sortedBy}%26count=${count}`, ``, ''], function(data) {
      if (data.results){
        that.setState({
          reviews: data.results.results
        });
      }
    });
  }

  getReviewsMetadata () {
    const that = this;
    this.props.getReviews([`reviews/meta?product_id=${this.props.product_id}`, ``, ''], function(data) {
      if (data.results){
        const recommendedPercentage = that.getRecommendedPercentage(data.results.recommended);
        that.setState({
          reviewsMetadata: data.results,
          recommendedPercentage: recommendedPercentage
        });
      }
    });
  }

  getRecommendedPercentage (recommended) {
    let sum = 0;
    for (const value in recommended) {
      sum += parseInt(recommended[value]);
    }
    const recommendedPercentage = Math.round((parseInt(recommended[true]) / sum) * 100);
    return recommendedPercentage;
  }

  updateSorting (event) {
    event.preventDefault();
    const sortedBy = event.target.value;
    this.setState({
      sortedBy: sortedBy
    }, () => {
      this.getReviews();
    });
  }

  updateRatingFilter (event) {
    const rating = parseInt(event.target.id);
    let filteredBy = this.state.filteredBy;
    if (filteredBy[rating]) {
      delete filteredBy[rating];
    } else {
      filteredBy[rating] = true;
    }

    this.setState({
      filteredBy: filteredBy
    });
  }

  removeRatingFilter () {
    this.setState({
      filteredBy: {}
    });
  }

  render () {
    const reviews = this.state.reviews;
    return (
      <div className="ratings-and-reviews">
        <Ratings
          reviewsMetadata={this.state.reviewsMetadata}
          product_id={this.props.product_id}
          recommendedPercentage={this.state.recommendedPercentage}
          updateRatingFilter={this.updateRatingFilter}
          filteredBy={this.state.filteredBy}
          removeRatingFilter={this.removeRatingFilter}/>
        <Reviews
          product_id={this.props.product_id}
          reviews={reviews}
          updateSorting={this.updateSorting}
          reviewsCharacteristics={this.state.reviewsMetadata.characteristics}
          filteredBy={this.state.filteredBy}/>
      </div>
    );
  }
}

RatingsWidget.propTypes = {
  getReviews: PropTypes.func,
  product_id: PropTypes.number
};

export default RatingsWidget;