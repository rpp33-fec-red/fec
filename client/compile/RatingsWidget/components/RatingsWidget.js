import React from 'react';
import '../ratings.scss';
import Reviews from './Reviews/Reviews.js';
import Ratings from './Ratings/Ratings.js';
import reviewsData from '../sample_data.js';

class RatingsWidget extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      sortedBy: 'relevance',
      filteredBy: {
        oneStar: false,
        twoStar: false,
        threeStar: false,
        fourStar: false,
        fiveStar: false,
      }
    }
    this.getData = this.getData.bind(this);
  }


  componentDidMount () {
    this.getData();
  }


  getData () {
    var options = {
      sort: 'newest'
    };
    this.props.getReviews('GET', ['reviews', 64620, ''], options, (data) => {
      console.log('these are the reviews:', data);
    });
  }


  render () {
    return (
      <div className="ratings-and-reviews">
        <Ratings/>
        <Reviews reviews={reviewsData.results} />
      </div>
    );
  }
}

export default RatingsWidget;