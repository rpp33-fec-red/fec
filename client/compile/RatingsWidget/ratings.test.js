import React from 'react';
import renderer from 'react-test-renderer';
import RatingsWidget from './components/RatingsWidget.js';

describe('Ratings widget renders correctly', () => {
  const tree = renderer
  .create(<RatingsWidget/>)
  .toJSON();
  expect(tree).toMatchSnapshot();
});