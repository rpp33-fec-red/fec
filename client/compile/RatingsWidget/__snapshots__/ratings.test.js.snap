// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ratings component renders correctly 1`] = `
<div
  className="ratings"
>
  <h4>
    Ratings & Reviews
  </h4>
  <div
    className="ratings-summary"
  >
    <h2>
      3.5
    </h2>
    <p
      className="star-rating"
    >
      * * * * *
    </p>
    <p>
      100% of the reviews recommend this product
    </p>
  </div>
  <div
    className="ratings-breakdown"
  >
    <div>
      5 stars ------------
    </div>
    <div>
      4 stars ------------
    </div>
    <div>
      3 stars ------------
    </div>
    <div>
      2 stars ------------
    </div>
    <div>
      1 stars ------------
    </div>
  </div>
  <div
    className="ratings-product-breakdown"
  >
    Ratings Product Breakdown Placeholder
  </div>
</div>
`;

exports[`RatingsWidget component renders correctly 1`] = `
<div
  className="ratings-and-reviews"
>
  <div
    className="ratings"
  >
    <h4>
      Ratings & Reviews
    </h4>
    <div
      className="ratings-summary"
    >
      <h2>
        3.5
      </h2>
      <p
        className="star-rating"
      >
        * * * * *
      </p>
      <p>
        100% of the reviews recommend this product
      </p>
    </div>
    <div
      className="ratings-breakdown"
    >
      <div>
        5 stars ------------
      </div>
      <div>
        4 stars ------------
      </div>
      <div>
        3 stars ------------
      </div>
      <div>
        2 stars ------------
      </div>
      <div>
        1 stars ------------
      </div>
    </div>
    <div
      className="ratings-product-breakdown"
    >
      Ratings Product Breakdown Placeholder
    </div>
  </div>
  <div
    className="reviews"
  >
    <div
      className="reviews-sorting"
    >
      <label
        htmlFor="sorting-option"
      >
        0
         reviews, sorted by 
      </label>
      <select
        name="sorting-option"
        onChange={[Function]}
      >
        <option
          value="relevant"
        >
          relevant
        </option>
        <option
          value="helpful"
        >
          helpful
        </option>
        <option
          value="newest"
        >
          newest
        </option>
      </select>
    </div>
    <div
      className="reviews-list"
    />
    <div
      className="reviews-buttons"
    >
      <button>
        ADD A REVIEW +
      </button>
    </div>
  </div>
</div>
`;

exports[`ReviewTile component renders correctly 1`] = `
<div
  className="review-tile"
>
  <div
    className="review-header"
  >
    <div
      className="star-rating"
    >
      * * * * *
    </div>
    <div
      className="right-corner"
    >
      shortandsweeet, April 13, 2019
    </div>
  </div>
  <h3
    className="review-summary"
  >
    I'm enjoying wearing these shades
  </h3>
  <div
    className="review-main"
  >
    <div
      className="review-body"
    >
      <p>
        Comfortable and practical.
      </p>
    </div>
    <div
      className="thumbnail-display"
    >
      <div>
        <img
          alt="Review photo 1 submitted by: shortandsweeet"
          className="review-image"
          id="shortandsweeet_1"
          onClick={[Function]}
          src="https://images.unsplash.com/photo-1560570803-7474c0f9af99?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=975&q=80"
        />
      </div>
      <div>
        <img
          alt="Review photo 2 submitted by: shortandsweeet"
          className="review-image"
          id="shortandsweeet_2"
          onClick={[Function]}
          src="https://images.unsplash.com/photo-1560570803-7474c0f9af99?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=975&q=80"
        />
      </div>
    </div>
    <div
      className="recommend"
    >
      â˜‘ I recommend this product
    </div>
    <div
      className="review-helpfulness-voting"
    >
      <p>
        Helpful? 
        <a
          onClick={[Function]}
        >
           Yes
        </a>
         (
        5
        )
      </p>
    </div>
  </div>
</div>
`;

exports[`Reviews component renders correctly 1`] = `
<div
  className="reviews"
>
  <div
    className="reviews-sorting"
  >
    <label
      htmlFor="sorting-option"
    >
      5
       reviews, sorted by 
    </label>
    <select
      name="sorting-option"
    >
      <option
        value="relevant"
      >
        relevant
      </option>
      <option
        value="helpful"
      >
        helpful
      </option>
      <option
        value="newest"
      >
        newest
      </option>
    </select>
  </div>
  <div
    className="reviews-list"
  />
  <div
    className="reviews-buttons"
  >
    <button
      onClick={[Function]}
    >
      MORE REVIEWS
    </button>
    <button>
      ADD A REVIEW +
    </button>
  </div>
</div>
`;

exports[`ReviewsList component renders correctly 1`] = `
Array [
  <div
    className="reviews-sorting"
  >
    <label
      htmlFor="sorting-option"
    >
      5
       reviews, sorted by 
    </label>
    <select
      name="sorting-option"
    >
      <option
        value="relevant"
      >
        relevant
      </option>
      <option
        value="helpful"
      >
        helpful
      </option>
      <option
        value="newest"
      >
        newest
      </option>
    </select>
  </div>,
  <div
    className="reviews-list"
  />,
  <div
    className="reviews-buttons"
  >
    <button
      onClick={[Function]}
    >
      MORE REVIEWS
    </button>
    <button>
      ADD A REVIEW +
    </button>
  </div>,
]
`;

exports[`ReviewsSorting component renders correctly 1`] = `
<div
  className="reviews-sorting"
>
  <label
    htmlFor="sorting-option"
  >
     reviews, sorted by 
  </label>
  <select
    name="sorting-option"
  >
    <option
      value="relevant"
    >
      relevant
    </option>
    <option
      value="helpful"
    >
      helpful
    </option>
    <option
      value="newest"
    >
      newest
    </option>
  </select>
</div>
`;
